### PR TITLE
[DOCS] Document cat allocation API response properties

### LINKED ISSUE #65523: [DOCS] Explanation for columns returned in _cat/allocation API
**Elasticsearch version** (`bin/elasticsearch --version`): N/A

**Plugins installed**: N/A

**JVM version** (`java -version`): N/A

**OS version** (`uname -a` if on a Unix-like system): N/A

**Description of the problem including expected versus actual behavior**: 

[The `GET _cat/allocation?v` API](https://www.elastic.co/guide/en/elasticsearch/reference/current/cat-allocation.html) currently shows the below columns:
```
shards disk.indices disk.used disk.avail disk.total disk.percent host      ip        node
     1         260b    47.3gb     43.4gb    100.7gb           46 127.0.0.1 127.0.0.1 CSUXak2
```

Though there is no concrete description of _exactly_ what each of these headings is measuring. Things like `shards`/`disk.total` are self explanatory, there appears to be some subtle difference between the values reported in `disk.indices` and `disk.used`. 

If we run `GET _cat/allocation?help` for some details we get:
```
shards       | s              | number of shards on node      
disk.indices | di,diskIndices | disk used by ES indices       
disk.used    | du,diskUsed    | disk used (total, not just ES)
disk.avail   | da,diskAvail   | disk available                
disk.total   | dt,diskTotal   | total capacity of all volumes 
disk.percent | dp,diskPercent | percent disk used             
host         | h              | host of node                  
ip           |                | ip of node                    
node         | n              | name of node
```

However, it's still 100% clear if `disk.indices` takes into consideration indices that have shards that are `UNASSIGNED`, for example nodes that have some data on disk for shards, but cannot allocate them due to tripping something like `cluster.routing.allocation.disk.watermark.*`. 

Checking [the source code](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/rest/action/cat/RestAllocationAction.java#L142), it appears as though this is indeed what's happening.

Can we update the docs for `GET _cat/allocation?v` to: 
1. Call out the above scenario of `disk.indices` vs `disk.used` in cases where shard data might exist on disk, but cannot be allocated due to circuit breakers, disk watermarks etc.
2. Document the columns like we do via `GET _cat/allocation?help`, in similar format to what we've done for [`GET _cat/nodes`](https://www.elastic.co/guide/en/elasticsearch/reference/current/cat-nodes.html)?

If we can get a concrete answer to the scenario mentioned in Item 1, then I'd be happy to raise a PR myself.

Thanks folks ❤️

### PR DESCRIPTION
**Pull request by jrodewig**
Documents the response properties for the cat allocation API.

Closes #65523

### Preview

https://elasticsearch_65635.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/cat-allocation.html#cat-allocation-api-response-body

### CODE REVIEW

File: docs/reference/cat/allocation.asciidoc [hunk 1]
```diff
@@ -40,6 +40,40 @@ include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=cat-s]
 
 include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=cat-v]
 
+[[cat-allocation-api-response-body]]
+==== {api-response-body-title}
+
+`shards`::
+Number of primary and replica shards assigned to the node.
```
Comment (jrodewig) [inline docs/reference/cat/allocation.asciidoc:47]: I intentionally omitted the data type here. cat APIs aren't intended for application consumption. Let me know if there's a use case I may be overlooking.

File: docs/reference/cat/allocation.asciidoc [hunk 2]
```diff
@@ -40,6 +40,40 @@ include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=cat-s]
...
+
+`disk.indices`::
+Disk space used by the node's shards. Does not include disk space for the
+<<index-modules-translog,translog>> or unassigned shards.
```
Comment (jrodewig) [inline docs/reference/cat/allocation.asciidoc:51]: Let me know if there are any other "gotchas" for `disk.indices`.

**Comment by DaveCTurner** [comment]
Ah sorry @jrodewig I somehow dropped this from my review list without reviewing it. It seems like a good change, did you mean to close it?

**Comment by jrodewig** [comment]
Hi @DaveCTurner. No worries at all. I was clearing out some of my stale PRs, and it looks like I got a little overzealous. 😄 

A review would be much appreciated!

File: docs/reference/cat/allocation.asciidoc [hunk 3]
```diff
@@ -40,6 +40,40 @@ include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=cat-s]
...
+applications or files on the node.
+
+`disk.avail`::
+Free disk space available to {es}.
+
+`disk.total`::
+Total disk space for the node, including in-use and available space.
+
+`disk.percent`::
+Total percentage of disk space in use. Includes {es}, the node's OS, and any
```
Comment (DaveCTurner) [inline docs/reference/cat/allocation.asciidoc:64]: Should we spell out that this is just `disk.used / disk.total`?

File: docs/reference/cat/allocation.asciidoc (hunk 1 above)
Comment (DaveCTurner) [reply docs/reference/cat/allocation.asciidoc:47]: :+1:

**Review by DaveCTurner — COMMENTED** [review]
Looks good, I left some suggestions.

File: docs/reference/cat/allocation.asciidoc [hunk 4]
```diff
@@ -40,6 +40,40 @@ include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=cat-s]
...
+`disk.indices`::
+Disk space used by the node's shards. Does not include disk space for the
+<<index-modules-translog,translog>> or unassigned shards.
+
+`disk.used`::
+Total disk space in use. Includes {es}, the node's OS, and any other
+applications or files on the node.
+
+`disk.avail`::
+Free disk space available to {es}.
```
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply docs/reference/cat/allocation.asciidoc:58]: Ah, sorry, correction: this is the number used for disk watermarks, not `disk.used`.

**Review by DaveCTurner — APPROVED** [review]
LGTM, thanks @jrodewig

**Comment by jrodewig** [comment]
Thanks @DaveCTurner! I appreciate the suggestions and review!

**Pull request merged**